### DISCUSSION TITLE
I have three user role author, editor and publisher .. how to do this in supabase. earlier disucssion did not help me.

### DISCUSSION THREAD

**Discussion by roker15** [supabase/supabase#3232] [Questions] [answered]
I have three user role **author, editor and publishe**r .. how to do this in supabase. earlier disucssion did not help me. auth.user has role **authenticated** only.

**Comment by awalias** [comment, marked as answer]
hey @roker15 thanks for the question but please refrain from spamming the discussion board, if useful you can refer to the answers on your previous question here: https://github.com/supabase/supabase/discussions/3129

you can implement this at the schema level as per the example linked, but our API gateway does not yet currently support custom "postgres level roles", this is being worked on and should be available in the coming months

**Reply by adimyth** [reply]
@awalias any updates on "custom roles at API gateway level"?

The solution mentioned in slack clone feels hacky & is not possible in real world to be true!
[Role-based access control (RBAC)](https://github.com/supabase/supabase/blob/master/examples/nextjs-slack-clone/README.md#role-based-access-control-rbac )

Are there any other solutions available?

**Reply by cgestes** [reply]
The part that create the roles from the email yeah it's just for the demo, but how to you manage role in supabase in this example seems fine to me, you just need your own logic to create the role, which is normal supabase code as far as I can tell.

**Comment by thisisbintang** [comment]


**Reply by amr3k** [reply]
Did you find anything @thisisbintang ?

**Reply by thisisbintang** [reply]
No

**Reply by 1saifj** [reply]
> No

And now?

**Reply by thisisbintang** [reply]
Still not yet

**Reply by 1saifj** [reply]
> Still not yet

that's bad

**Reply by jsgill** [reply]
i was looking for set custom roles at my application as admin, advertiser & user. i didn't find any solution in supabase, so used firebase for role authentication and rest of the handling at supabase.

**Reply by starascendin** [reply]
> i was looking for set custom roles at my application as admin, advertiser & user. i didn't find any solution in supabase, so used firebase for role authentication and rest of the handling at supabase.

can you share how you are doing this?

**Reply by tolgahanbora** [reply]
> i was looking for set custom roles at my application as admin, advertiser & user. i didn't find any solution in supabase, so used firebase for role authentication and rest of the handling at supabase.

That's too confused bro..

**Reply by steve-chavez** [reply]
> i was looking for set custom roles at my application as admin, advertiser & user.

See https://github.com/orgs/supabase/discussions/11948#discussioncomment-4802137
